addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js

```diff
@@ -0,0 +1,182 @@
+        switch (type) {
+            case "many2many":
+            case "tags":
+                return ["in", "not in", "set", "not_set"];
+            case "many2one":
+            case "selection":
+                return ["=", "!=", "set", "not_set"];
+            default:
+                return ["set", "not_set"];
+        }
+    }
+
+    switch (type) {
+        case "boolean":
+            return ["is", "is_not"];
+        case "selection":
+            return ["=", "!=", "in", "not in", "set", "not_set"];
+        case "char":
+        case "text":
+        case "html":
+            return ["=", "!=", "ilike", "not ilike", "in", "not in", "set", "not_set"];
+        case "date":
+        case "datetime":
+            return ["=", "!=", ">", ">=", "<", "<=", "between", "set", "not_set"];
+        case "integer":
+        case "float":
+        case "monetary":
+            return [
+                "=",
+                "!=",
+                ">",
+                ">=",
+                "<",
+                "<=",
+                "between",
+                "ilike",
+                "not ilike",
+                "set",
+                "not_set",
+            ];
+        case "many2one":
+        case "many2many":
+        case "one2many":
+            return ["in", "not in", "=", "!=", "ilike", "not ilike", "set", "not_set"];
+        case "json":
+            return ["=", "!=", "ilike", "not ilike", "set", "not_set"];
+        case undefined:
+            return ["="];
+        default:
+            return [
+                "=",
+                "!=",
+                ">",
+                ">=",
+                "<",
+                "<=",
+                "ilike",
+                "not ilike",
+                "like",
+                "not like",
+                "=like",
+                "=ilike",
+                "child_of",
+                "parent_of",
+                "in",
+                "not in",
+                "set",
+                "not_set",
+            ];
+    }
+}
+
+function toKey(operator, negate = false) {
+    if (!negate && typeof operator === "string" && operator in OPERATOR_DESCRIPTIONS) {
+        // this case is the main one. We keep it simple
+        return operator;
+    }
+    return JSON.stringify([formatValue(operator), negate]);
+}
+
+function toOperator(key) {
+    if (!key.includes("[")) {
+        return [key, false];
+    }
+    const [expr, negate] = JSON.parse(key);
+    return [toValue(parseExpr(expr)), negate];
+}
+
+export function getOperatorLabel(operator, negate = false) {
+    let label;
+    if (typeof operator === "string" && operator in OPERATOR_DESCRIPTIONS) {
+        if (negate && operator in TERM_OPERATORS_NEGATION) {
+            return OPERATOR_DESCRIPTIONS[TERM_OPERATORS_NEGATION[operator]];
+        }
+        label = OPERATOR_DESCRIPTIONS[operator];
+    } else {
+        label = formatValue(operator);
+    }
+    if (negate) {
+        return sprintf(`not %s`, label);
```

Comment:
```suggestion
        return `not ${label}`;
```

Reply:
remove sprintf everywhere and use _t

Reply:
import * as dsHelpers from ... (shorter)

Reply:
done